service/src/main/java/tds/assessment/repositories/AdminSubjectQueryRepository.java

```diff
@@ -0,0 +1,18 @@
+package tds.assessment.repositories;
+
+import java.util.Optional;
+
+import tds.assessment.SetOfAdminSubject;
+
+/**
+ * Queries for admin subject related entities
+ */
+public interface AdminSubjectQueryRepository {
+    /**
+     * Finds a {@link tds.assessment.SetOfAdminSubject} by key
+     *
+     * @param adminSubjectKey the unique key for the admin subject
+     * @return {@link tds.assessment.SetOfAdminSubject} if found otherwise empty
+     */
+    Optional<SetOfAdminSubject> findByKey(String adminSubjectKey);
```

Comment:
Does this mean there was consensus on findXXX returns Optional while getXXX returns class directly?